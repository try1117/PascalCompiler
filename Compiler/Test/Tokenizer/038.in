// sum two numbers
function sum(a, b: integer) : integer;
begin
  sum := a + b; // a + b
  (* a + b =
  b + a*)
end;
