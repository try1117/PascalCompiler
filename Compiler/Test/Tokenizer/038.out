2  | 1  | KEYWORD_FUNCTION         | function                 | 
2  | 10 | IDENTIFIER               | sum                      | 
2  | 13 | SEP_BRACKET_LEFT         | (                        | 
2  | 14 | IDENTIFIER               | a                        | 
2  | 15 | SEP_COMMA                | ,                        | 
2  | 17 | IDENTIFIER               | b                        | 
2  | 18 | OP_COLON                 | :                        | 
2  | 20 | KEYWORD_INTEGER          | integer                  | 
2  | 27 | SEP_BRACKET_RIGHT        | )                        | 
2  | 29 | OP_COLON                 | :                        | 
2  | 31 | KEYWORD_INTEGER          | integer                  | 
2  | 38 | SEP_SEMICOLON            | ;                        | 
3  | 1  | KEYWORD_BEGIN            | begin                    | 
4  | 3  | IDENTIFIER               | sum                      | 
4  | 7  | KEYWORD_ASSIGN           | :=                       | 
4  | 10 | IDENTIFIER               | a                        | 
4  | 12 | OP_PLUS                  | +                        | 
4  | 14 | IDENTIFIER               | b                        | 
4  | 15 | SEP_SEMICOLON            | ;                        | 
7  | 1  | KEYWORD_END              | end                      | 
7  | 4  | SEP_SEMICOLON            | ;                        | 
